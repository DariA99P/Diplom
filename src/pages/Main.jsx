import React, { Component } from 'react';

import {
  BrowserRouter as Router,
  Route,
  Switch,
} from 'react-router-dom';

import { Layout } from 'antd';

import Header from '../layout/headerComponent/HeaderComponent';
import Footer from '../layout/footerComponent/FooterComponent';
import Sidebar from '../layout/sidebarComponent/SidebarComponent';

import Home from './home/Home';
import NotFound from './notFound/NotFound';

import '../assets/styles/styles.css';

const { Content } = Layout;

class Main extends Component {
  state = {
    collapsed: false,
  }

  toggleSidebar = () => {
    const { collapsed } = this.state;

    this.setState({
      collapsed: !collapsed,
    });
  }

  render() {
    const { collapsed } = this.state;
    return (
      <Router>
        <Layout style={{ minHeight: '100vh' }}>
          <Sidebar collapsed={collapsed} toggle={this.toggleSidebar} />
          <Layout>
            <Header collapsed={collapsed} toggle={this.toggleSidebar} />
            <Content style={{ margin: '16px 16px 0 16px' }}>
              <Switch>
                <Route exact path="/" component={Home} />
                <Route component={NotFound} />
              </Switch>
            </Content>
            <Footer />
          </Layout>
        </Layout>
      </Router>
    );
  }
}

export default Main;
