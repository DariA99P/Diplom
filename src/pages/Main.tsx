import React, { Component } from 'react';

import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import { Layout } from 'antd';

import Header from '../layout/headerComponent/HeaderComponent';
import Footer from '../layout/footerComponent/FooterComponent';
import Sidebar from '../layout/sidebarComponent/SidebarComponent';

// import Home from './home/Home';
import NotFound from './notFound/NotFound';

import '../assets/styles/styles.css';
import PreparationTasks from './preparationTasks';

const { Content } = Layout;

class Main extends Component {
  state = {
    collapsed: false,
  };

  toggleSidebar = () => {
    const { collapsed } = this.state;

    this.setState({
      collapsed: !collapsed,
    });
  };

  render() {
    const { collapsed } = this.state;
    return (
      <Router>
        <Layout style={{ minHeight: '100vh' }}>
          <Sidebar collapsed={collapsed} toggle={this.toggleSidebar} />
          <Layout>
            <Header collapsed={collapsed} toggle={this.toggleSidebar} />
            <Content>
              <Switch>
                <Route exact path="/" component={PreparationTasks} />
                <Route component={NotFound} />
              </Switch>
            </Content>
            <Footer />
          </Layout>
        </Layout>
      </Router>
    );
  }
}

export default Main;
