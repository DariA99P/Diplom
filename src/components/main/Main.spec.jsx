import React from 'react';
import { shallow } from 'enzyme';

import Main from '.';

describe('<Main />', () => {
  let tree;
  beforeEach(() => {
    tree = shallow(<Main />);
  });

  describe('On rendering', () => {
    it('should renders properly', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Handling toggle state changes', () => {
    it('should update state properly', () => {
      const previousState = tree.state('collapsed');
      tree.instance().toggleSidebar();

      expect(tree.state('collapsed')).toEqual(!previousState);
    });
  });
});
