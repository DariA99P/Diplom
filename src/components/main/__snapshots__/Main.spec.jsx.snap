// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> On rendering should renders properly 1`] = `
<Adapter
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <SideBar
    collapsed={false}
    toggle={[Function]}
  />
  <Adapter>
    <Connect(Header)
      collapsed={false}
      toggle={[Function]}
    />
    <ActionBar />
    <Content>
      <Wrapper>
        <Routes />
      </Wrapper>
    </Content>
    <span
      className="copyright"
    >
      Copyright Â© 2018 VistaJet
    </span>
  </Adapter>
</Adapter>
`;
